pkg/openstack/instances.go

```diff
@@ -633,6 +633,44 @@ func nodeAddresses(srv *servers.Server, ports []ports.Port, networkingOpts Netwo
 		return nil, err
 	}
 
+	// Add the addresses assigned on subports via trunk
+	// This exposes the vlan networks to which subports are attached
+	subportAddresses := make(map[string][]Address)
+
+	for _, port := range ports {
+		for _, subport := range port.TrunkDetails.SubPorts {
+			p, err := neutronports.Get(client, subport.PortID).Extract()
+			if err != nil {
```

Comment:
Nice! This only has a cost if there are trunk ports. It is free on systems which are not using trunk ports.